src/anonymizer.js

```diff
@@ -259,13 +259,15 @@ export function cleanTags(
             var tagNumber = tagInfo.tag,
                 tagString = Tag.fromPString(tagNumber).toCleanString();
             if (dict[tagString]) {
-                var newValue;
-                if (tagString in tagNamesToReplace) {
-                    newValue = [tagNamesToReplace[tagString]];
+                if (Array.isArray(dict[tagString].Value)) {
```

Comment:
I don't think this is the right behaviour because it modifies the original string, which is going to cause problems.  You do NOT want to keep the references here because of that.  Also, you don't want to clear every single array element there is, only the ones which have tag names to replace.  The other ones should be left alone.